transcom/mymove
# [MB-6194] Extend user endpoint to update Active status

## Description

This PR updates the existing endpoint `PATCH /users/{userId}` to support changing the `Active` status of a user in the `users` table.

* The previous iteration used the update endpoint only to revoke user sessions, which was reflected in the naming.
* I've renamed all related functions, etc. to `UpdateUser`, `UserUpdatePayload`, etc. None of the previous code by @monfresh was altered except for renaming.
* This makes the code more flexible in case we need to add additional user properties to update in the future.
* Added new service object to handle updating the user (aside from session revocation, which already has its own SO). For now, this only handles the Active status, but I left a comment to direct any future engineer to add their changes to `UpdateUser`.
* Updated tests for the handler and added tests for the service object

## Reviewer Notes

See: https://github.com/transcom/mymove/compare/update-user-in-admin-app-mb-6194?expand=1#diff-1c02fbb6907330dee84754ccc54da0c2f62468c90e66e838504434b740df4279R126

Would love to know if there's a better way to do this in golang!

## Setup

Tests:

`go test ./pkg/services/user --testify.m "TestUserUpdater" -v`
`go test ./pkg/handlers/adminapi/ --testify.m "TestUpdateUserHandler" -v`

You can also test the endpoint using Postman if you would like to brave the CSRF token setup. YOU CAN DO IT. Ping me if you want to pair to set it up.

Here are some instructions I wrote down for myself:

* Make sure you have Postman Interceptor installed in your browser and you've added `adminlocal` as a domain
* Open Postman -> Make sure Interceptor is turned on in Postman and you've added `adminlocal` as a domain
* Run `make server_run` and `make admin_client_run`
* In admin app, log in as local admin user
* Open up dev tools -> Application -> Cookies -> Copy the token for `admin_session`
* In Postman, make GET request to `http://adminlocal:3000/admin/v1/users/{userId}` You'll probably get a 403 because we need to update the admin_session token.
* Click the `cookies` link in your postman app (underneath the request path) -- you should see all populated. Click the `admin_session` cookie, paste in your new value, and save.
* You can redo the GET request now and you should receive a response.
* To do a `PATCH` request, click the `cookies` link again and copy the `masked_gorilla` token value.
* Add a header for your request with the key `X-CSRF-TOKEN` and value `<token value you copied>`
* Add a body to toggle the `active` status.
```
{ "active": false}
```
* Hit it!


## Code Review Verification Steps

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-6194) for this change
